### ISSUE TITLE
There is a way to put an intercept or middleware on the Router

### ISSUE THREAD

**Issue by insinfo** [dart-lang/angular#1867]
There is a way to put an intercept or middleware on the Router to prevent unauthorized access without login, because having to implement a life cycle on each component to prevent unauthorized access is too much work.
There could be a way to do this similar to the Slim Framework, in Slim you create Authentication Middleware and link this Middleware to a route group.

* [x] Dart SDK 2.5.2
* [x] AngularDart 5.3.0
* [x] Windows
* [x] Chrome

**Comment by leonsenft** [comment]
Have you tried using RouterHook?

https://github.com/dart-lang/angular/blob/44d8ee96f03e54fadb1fbdf0e021570f8d88f6da/angular_router/lib/src/router_hook.dart#L85-L110

**Comment by insinfo** [comment]
@leonsenft    
Forgive me, but I couldn't understand how to use "RouterHook" in a way that I can redirect to a specific route if the user is not logged in, without having to modify a large component container.

**Comment by leonsenft** [comment]
Yes, that capability is certainly lacking. Same goes for all of the route guards unfortunately. I think the API of can{PerformAction} returning true/false is flawed. Ideally the return value would also accept a route/url to allow redirection.

I think the only workaround now is to trigger a navigation to the route you desire, then return false from the hook. Of course injecting Router into RouterHook is tricky since it forms a dependency cycle, but this can be avoided by injecting it lazily. See this test for inspiration:

https://github.com/dart-lang/angular/blob/44d8ee96f03e54fadb1fbdf0e021570f8d88f6da/angular_router/test/router_hook_test.dart#L200-L202
